R/mapbiomas.R

```diff
@@ -0,0 +1,277 @@
+    tab<-a
+    if(!is.null(code_state)){
+      tab<-tab[!(tab$state!=code_state),]
+    }
+    if(!is.null(covering)){
+      tab<-tab[!(substring(tab$level_1,1,1)!=covering),]
+    }
+    if(type=='stacked' | type=='empilhado'){
+      for(i in year_begin:year_end){
+        ret<-c()
+        ret<-tab[,1:8]
+        ano<-as.character(i)
+        ret<-cbind(ret,tab[,ano])
+        colnames(ret)[which(colnames(ret) == ano)] <- 'Area'
+        num<-nrow(tab[,ano])
+        data<-rep(c(i),num)
+        ret<-cbind(ret,data)
+        retorno<-rbind(retorno,ret)
+      }
+    }
+    if(type=='normal'){
+      ret<-tab[,1:8]
+      for(i in year_begin:year_end){
+        ano<-as.character(i)
+        ret<-cbind(ret,tab[,ano])
+      }
+      retorno<-ret
+    }
+  }else if(aggregation_level=='municipality' | aggregation_level=='municipio'){
+    b<-read_excel(path = p1f, sheet = 3)
+    tipos<-c('RR','RO','AC','AM','MA','TO','PA','AP','MT')
+    `%notin%` <- Negate(`%in%`)
+    b<-b[!(b$state %notin% as.vector(tipos)),]
+    retorno<-data.frame()
+    tab<-b
+    if(!is.null(code_state)){
+      tab<-tab[!(tab$state!=code_state),]
+    }
+    if(!is.null(code_mun)){
+      tab<-tab[!(tab$municipality!=code_mun),]
+    }
+    if(!is.null(covering)){
+      tab<-tab[!(substring(tab$level_1,1,1)!=covering),]
+    }
+    if(type=='stacked' | type=='empilhado'){
+      for(i in year_begin:year_end){
+        ret<-c()
+        ret<-tab[,1:8]
+        ano<-as.character(i)
+        ret<-cbind(ret,tab[,ano])
+        colnames(ret)[which(colnames(ret) == ano)] <- 'Area'
+        num<-nrow(tab[,ano])
+        data<-rep(c(i),num)
+        ret<-cbind(ret,data)
+        retorno<-rbind(retorno,ret)
+      }
+    }
+    if(type=='normal'){
+      ret<-tab[,1:8]
+      for(i in year_begin:year_end){
+        ano<-as.character(i)
+        ret<-cbind(ret,tab[,ano])
+      }
+      retorno<-ret
+    }
+  }
+  return(retorno)
+}
+
+#' @title load_mapbiomas_transicao
+#' 
+#' @description Download and filter data on transition of types of soil covering by year
+#' 
+#' @param path A string indicating where the raw data is in your computer. The default is NULL which means the data will be extracted directly from the website
+#' @param code_state Output contains only data from the state selected
+#' Input has to be IBGE's coding for the state desired
+#' @param covering_from Output contains only data of the transition from the covering selected to all other coverings
+#' Input has to be the code of the desired covering
+#' @param covering_to Output contains only data of the transition from all coverings to the covering selected
+#' Input has to be the code of the desired covering
+#' @param type Decide if the output should have a column for each year (normal) or a single column for the areas of all years (stacked | empilhado)
+#' @param year_diff A numeric object containing the year differential desired in the data base, or have all included (NULL)
```

Comment:
Não entendi exatamente o que está acontecendo com o `year_diff`. Acho que vale documentar melhor

Reply:
Os dados de transição existem em diferentes intervalos de tempo. EX.: 2000-2005; 2005-2010; 2010-2015 e também 1990-2000; 2000-2010; 2010-2020, entre outros. O year_diff só filtra isso se a pessoa quiser os dados com transições anuais, de dois em dois anos, de cinco em cinco etc.

Reply:
No dataset mesmo eles disponibilizam em todos esses intervalos? Não tem nenhuma operação para "agregar no tempo"?

Reply:
Hmm, entendi. Eu chamaria o parâmetro então de "transition interval" ou alguma coisa assim e colocaria as opções disponíveis tipo: 
transition_interval One of {1, 2, 5, 10} specifying the desired interval in years to observe transitions ie choosing 1 will retrieve yearly transition data.